Panic with keyvalue type redis and redis not enabled

### Pre-submission checklist

- [x] I have searched for existing open or closed issue reports with the same problem.

### Description

During initial setup, i set the keyvalue type to redis, but missed adding enabled to redis, even though the configuration was there. That ended up with this panic
```
time=2026-04-12T07:18:46.845Z level=INFO msg="Using config file: /etc/vikunja/config.yml"
time=2026-04-12T07:18:46.845Z level=INFO msg="Running migrations…"
time=2026-04-12T07:18:46.909Z level=INFO msg="Ran all migrations successfully."
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x103e232]

goroutine 1 [running]:
code.vikunja.io/api/pkg/modules/keyvalue/redis.(*Storage).GetWithValue(0x1?, {0x251fbc4, 0x10}, {0x20e84c0, 0xc0007ebb48})
	/go/src/code.vikunja.io/api/pkg/modules/keyvalue/redis/redis.go:82 +0x32
code.vikunja.io/api/pkg/modules/keyvalue.GetWithValue(...)
	/go/src/code.vikunja.io/api/pkg/modules/keyvalue/keyvalue.go:62
code.vikunja.io/api/pkg/modules/auth/openid.GetAllProviders()
	/go/src/code.vikunja.io/api/pkg/modules/auth/openid/providers.go:77 +0xa5
code.vikunja.io/api/pkg/initialize.FullInitWithoutAsync()
	/go/src/code.vikunja.io/api/pkg/initialize/init.go:103 +0x79
code.vikunja.io/api/pkg/initialize.FullInit()
	/go/src/code.vikunja.io/api/pkg/initialize/init.go:121 +0x17
code.vikunja.io/api/pkg/cmd.init.func36(0xc00035d900?, {0x24e34dd?, 0x4?, 0x24e347d?})
	/go/src/code.vikunja.io/api/pkg/cmd/web.go:140 +0xf
github.com/spf13/cobra.(*Command).execute(0x42df340, {0xc000040090, 0x0, 0x0})
	/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1004 +0xa5a
github.com/spf13/cobra.(*Command).ExecuteC(0x42df340)
	/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1071
code.vikunja.io/api/pkg/cmd.Execute()
	/go/src/code.vikunja.io/api/pkg/cmd/cmd.go:44 +0x1a
main.main()
	/go/src/code.vikunja.io/api/main.go:22 +0xf
```

Setting `.redis.enabled: true` addressed the issue. While not a big problem, this makes it a bit more complicated for users to know what part was wrong

### Vikunja Version

2.3.0

### Browser and version

_No response_

### Can you reproduce the bug on the Vikunja demo site?

No

### Screenshots

_No response_

--- a/pkg/modules/keyvalue/keyvalue.go
+++ b/pkg/modules/keyvalue/keyvalue.go
@@ -18,6 +18,7 @@ package keyvalue
 
 import (
 	"code.vikunja.io/api/pkg/config"
+	"code.vikunja.io/api/pkg/log"
 	"code.vikunja.io/api/pkg/modules/keyvalue/memory"
 	"code.vikunja.io/api/pkg/modules/keyvalue/redis"
 )
@@ -40,6 +41,9 @@ var store Storage
 func InitStorage() {
 	switch config.KeyvalueType.GetString() {
 	case "redis":
+		if !config.RedisEnabled.GetBool() {
+			log.Fatalf("keyvalue.type is set to %q but redis is not enabled. Please set redis.enabled to true in your configuration.", config.KeyvalueType.GetString())
+		}
 		store = redis.NewStorage()
 	case "memory":
 		fallthrough

